### PR TITLE
feat(gql): 1778 allows file size unit

### PR DESCRIPTION
**Pull request by ithinkicancode**
Closes #1778

### CODE REVIEW

File: gql.pegjs [hunk 1]
```diff
@@ -33,12 +33,47 @@
 // grammar GQL
 
 {
+
+  function endsWith(subjectString, searchString, position) {
```
Comment (shanewilson) [inline gql.pegjs]: can you make this part of the grammar instead of post processing?

Comment (ithinkicancode) [inline gql.pegjs]: The whole thing (translate the size units to bytes) or just `endsWith`?

**Comment by cy** [comment]
Got an API exception for ZB and YB, otherwise 👍 

```
NumberFormatException[For input string: "1.1805916207174113e+21"];
```

**Comment by ithinkicancode** [comment]
@cy, thanks. Removed ZB & YB as they are most likely not in use (EB is probably enough).

**Comment by phuongmy** [comment]
Talked with Terry: UI 1.1.2 @ f200016, API 1.1.4 @ a02c407 did not have the change about ZB&YB.
For the new iteration, this PR should include removal of ZB, YB, EB, i.e do not accept these values in the string and display invalid query accordingly so that it does not create an exception.

Reminder for testing: `files.file_size < "5GB"`

**Comment by phuongmy** [comment]
This query `files.file_size <= "13 GB" ` returns files with a size > 13GB.
screen shot 2016-07-11 at 11 25 03 am

**Comment by phuongmy** [comment]
These queries:
` files.file_size = "13 GB" ` 
` files.file_size != "13 GB" ` 
throw an exception.
screen shot 2016-07-11 at 11 28 20 am

**Comment by alex-wilmer** [comment]
@phuongmy This is ready for re-testing

note file size values do not require strings. `files.file_size < 13GB` returns the correct files now.

**Comment by alex-wilmer** [comment]
@shanewilson r?

**Comment by cy** [comment]
```files.file_size = 13GB```, ```http://localhost:5000/cases?expand=summary.data_categories&fields=case_id,annotations.annotation_id,project.project_id,project.name,project.primary_site,project.program.name,project.disease_type,submitter_id,demographic.gender,demographic.race,demographic.ethnicity,diagnoses.primary_diagnosis,diagnoses.vital_status,diagnoses.days_to_death,diagnoses.age_at_diagnosis&filters={"op":"and","content":[{"op":"=","content":{"field":"files.file_size","value":13000000000}}]}&from=1&size=20&sort=case_id:asc``` gives
```
  File "/Users/CYu/repos/portal-api/gdcapi/resources/index/v0.py", line 145, in get_cases
    response = format_response(request_options, cases.search(request_options), mimetype)
  File "/Users/CYu/repos/portal-api/gdcapi/services/cases.py", line 9, in search
    return cases.search(params=params)
  File "/Users/CYu/repos/portal-api/gdcapi/esutils/search.py", line 256, in search
    body = build_body(doc_type, nested_fields, sd_params)
  File "/Users/CYu/repos/portal-api/gdcapi/esutils/search.py", line 93, in build_body
    f = request.build_filters(doc_type, nested_fields, filters)
  File "/Users/CYu/repos/portal-api/gdcapi/esutils/request.py", line 478, in build_filters
    f = _get_and_filter(doc_type, nested, content)
  File "/Users/CYu/repos/portal-api/gdcapi/esutils/request.py", line 320, in _get_and_filter
    curr = _get_term_or_prefix_filter(doc_type, nested, x)
  File "/Users/CYu/repos/portal-api/gdcapi/esutils/request.py", line 123, in _get_term_or_prefix_filter
    if '*' in value[0]:
TypeError: 'int' object has no attribute '__getitem__'```

same with ```files.file_size != 13GB``` other operators fine and dandy

**Comment by alex-wilmer** [comment]
@cy fixed by https://github.com/NCI-GDC/gdcapi/pull/556

**Pull request merged**